lib/puppet/type/cisco_ospf_area_vlink.rb

```diff
@@ -0,0 +1,248 @@
+    identity = ->(x) { x }
+    patterns = []
+
+    # Below pattern matches both parts of the full composite name.
+    patterns << [
+      /^(\S+) (\S+) (\S+) (\S+)$/,
+      [
+        [:ospf, identity],
+        [:vrf, identity],
+        [:area, identity],
+        [:vlink, identity],
+      ],
+    ]
+    patterns
+  end
+
+  # Overwrites name method. Original method simply returns self[:name],
+  # which is no longer valid or complete.
+  # Would not have failed, but just return nothing useful.
+  def name
+    "#{self[:ospf]} #{self[:vrf]} #{self[:area]} #{self[:vlink]}"
+  end
+
+  newparam(:name) do
+    desc 'Name of cisco_ospf_area_vlink, not used, but needed for puppet'
+  end
+
+  newparam(:vlink, namevar: true) do
+    desc 'Name of the virtual_link instance. Valid values are string.'
+  end # param vlink
+
+  newparam(:area, namevar: true) do
+    desc 'Name of the resource instance. Valid values are string.'
+    munge do |value|
+      value = IPAddr.new(value.to_i, Socket::AF_INET) unless
+        value[/\./]
+      value
+    end
+  end # param area
+
+  newparam(:vrf, namevar: true) do
+    desc "Name of the vrf instance. Valid values are string. The
+          name 'default' is a valid VRF."
+  end # param vrf
+
+  newparam(:ospf, namevar: true) do
+    desc 'Name of the ospf instance. Valid values are string.'
+  end # param ospf
+
+  ##############
+  # Attributes #
+  ##############
+
+  newproperty(:auth_key_chain) do
+    desc "Authentication password key chain name. Valid
+          values are string, keyword 'default'. "
+
+    munge { |value| value == 'default' ? :default : value }
+  end # property auth_key_chain
+
+  newproperty(:authentication) do
+    desc 'Enable authentication for the virtual_link.'
+
+    newvalues(:cleartext, :md5, :null, :default)
+  end # property authentication
+
+  newproperty(:authentication_key_encryption_type) do
+    desc "Specifies the scheme used for encrypting
+          authentication key password. Valid values are
+          'cleartext', '3des' or 'cisco_type_7' encryption,
+          and 'default', which defaults to 'cleartext'."
+
+    newvalues(:cleartext,
+              :'3des',
+              :cisco_type_7,
+              :default)
+
+    munge do |value|
+      value = :cleartext if value.to_sym == :default
+      value.to_sym
+    end
+  end # property authentication_key_encryption_type
+
+  newproperty(:authentication_key_password) do
+    desc "Specifies the authentication key password. Valid values are
+          string, keyword 'default'"
+
+    munge { |value| value == 'default' ? :default : value }
+  end # property authentication_key_password
+
+  newproperty(:dead_interval) do
+    desc "Sets the time in seconds that a neighbor waits for a Hello packet
+          before declaring the local router as dead and tearing down
+          adjacencies. Valid values are integer, keyword 'default'."
+
+    munge { |value| value == 'default' ? :default : Integer(value) }
```

Comment:
The NU is returning int type and so need to convert this to Integer if a string is given in the manifest.